### ISSUE TITLE
Can I use it with html5ever?

### ISSUE THREAD

**Issue by LeMoussel** [shepmaster/sxd-xpath#73]
Can I use sxd-xpath with [html5ever](https://crates.io/crates/html5ever) to find node in HTML document?

For example:
let html_file =  Path::new("cocon1.html").to_path_buf();
```
let html_file =  Path::new("cocon1.html").to_path_buf();
let dom = parse_document(RcDom::default(), Default::default())
	            .from_utf8()
	            .from_file(&html_file)
	            .unwrap();
let html_content = document.to_string();
```

**Comment by shepmaster** [comment]
No, but this could be interesting to investigate. I don't think it would be very easy though, as I believe (but have not verified) that the underlying access to the DOMs is not the most similar.

**Comment by mildred** [comment]
Does sxd-xpath depends directly on a sxd(document object or does it depends on a trait which is implemented using sxd-document but can be re-implemented for any parser with similar access? If not, is that complicated to do so?

**Comment by shepmaster** [comment]
> Does sxd-xpath depends directly on a sxd-document object

Yes, basically everywhere uses a concrete `sxd_xpath::nodeset::Node` which itself refers to 7 concrete types from sxd-document. The code then matches on that in various places and uses methods on those types. The `Node` is also a public API.

I don't have an answer for "is it complicated". My *gut feeling* is that it would be, but sometimes the only way to find out is to just try.

**Comment by carlolefou** [comment]
@shepmaster I needed to evaluate xpaths against html for scraping so I wrote this helper library to bridge html5ever with sxd-document. This is pretty much my first real rust code so, if you have some time, please take a look and send any feedback my way. thanks! https://github.com/carlolefou/sxd-html
